### PR TITLE
Add create build occurrence action

### PR DESCRIPTION
**Pull request by alexashley**
It's worth noting that the logs are scoped to the current job, not the entire workflow, so the action assumes it's running as a step in the build job. It's possible to change that scope to the entire workflow, but we lose out on easy access to logs (AFAICT you can only get a ZIP of workflow logs). 

Here's a [sample run](https://github.com/rode/demo-app/runs/2521952334?check_suite_focus=true) against the a branch in the `demo-app` repo.

<details>
<summary>And the output:</summary>

```json
{
  "name": "projects/rode/occurrences/c3a09722-cd2d-4b83-8526-309a5b4c4404",
  "resource": {
    "name": "",
    "uri": "git://github.com/rode/demo-app@a47eaf283cedd247f4dc674af27ac5f9ac36f510",
    "contentHash": null
  },
  "noteName": "projects/rode/notes/build_collector",
  "kind": "BUILD",
  "remediation": "",
  "createTime": "2021-05-06T20:45:20.844903806Z",
  "updateTime": null,
  "build": {
    "provenance": {
      "id": "https://github.com/rode/demo-app/runs/2521952334",
      "projectId": "projects/rode",
      "commands": [],
      "builtArtifacts": [
        {
          "checksum": "",
          "id": "harbor.internal.lead.prod.liatr.io/rode-demo/test-action@sha256:1234",
          "names": [
            "harbor.internal.lead.prod.liatr.io/rode-demo/test-action:v1.2.3"
          ]
        }
      ],
      "createTime": "2021-05-06T20:45:20.822890153Z",
      "startTime": "2021-05-06T20:40:49Z",
      "endTime": "2021-05-06T20:45:20.821925569Z",
      "creator": "alexashley",
      "logsUri": "https://github.com/rode/demo-app/commit/a47eaf283cedd247f4dc674af27ac5f9ac36f510/checks/2521952334/logs",
      "sourceProvenance": {
        "artifactStorageSourceUri": "",
        "fileHashes": {},
        "context": {
          "git": {
            "url": "https://github.com/rode/demo-app/commit/a47eaf283cedd247f4dc674af27ac5f9ac36f510",
            "revisionId": "a47eaf283cedd247f4dc674af27ac5f9ac36f510"
          },
          "labels": {}
        },
        "additionalContexts": []
      },
      "triggerId": "",
      "buildOptions": {},
      "builderVersion": ""
    },
    "provenanceBytes": ""
  }
}
```
</details>

### CODE REVIEW

File: action.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+name: create-build-occurrence
+author: rode@liatrio.com
+description: Creates a build occurrence in Rode
+branding:
+  icon: anchor
+  color: green
+
+runs:
+  using: docker
+  image: Dockerfile
```
Comment (alexashley) [inline action.yaml:10]: Once there's a release, I'll switch this to reference an image.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,21 @@
+FROM golang:1.16-alpine as builder
+
+WORKDIR /workspace
+
+RUN apk add --no-cache git
```
Comment (mrparkers) [inline Dockerfile:5]: I'm guessing the `github` package needs `git` installed for the builder image?

**Review by mrparkers — APPROVED** [review]

File: Dockerfile (hunk 2 above)
Comment (mrparkers) [reply Dockerfile]: ...or not, I see you removed it 😄

Comment (alexashley) [reply Dockerfile]: Just a copypasta error on my part, removed.

**Pull request merged**